src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/yarn/YarnPackager.groovy

```diff
@@ -45,14 +45,14 @@ class YarnPackager {
     @Autowired
     NameVersionNodeTransformer nameVersionNodeTransformer
 
-    public DependencyGraph parse(String yarnLockText) {
+    public DependencyGraph parse(List<String> yarnLockText) {
         def rootNode = new NameVersionNodeImpl()
         rootNode.name = "detectRootNode - ${UUID.randomUUID()}"
         def nameVersionLinkNodeBuilder = new LinkedNameVersionNodeBuilder(rootNode)
 
         NameVersionNode currentNode = null
         boolean dependenciesStarted = false
-        for (String line : yarnLockText.split(System.lineSeparator())) {
+        for (String line : yarnLockText) {
             if (!line.trim()) {
```

Comment:
Why dont you trim the line and use that rather than trimming it here and again later on line 66?